addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf

```diff
@@ -5,188 +5,173 @@
+    // determine how many wounds to create
+    private _nWounds = [_damage, _thresholds, true] call FUNC(interpolatePoints);
+    if (_nWounds < 1) then {
+        TRACE_2("Damage created zero wounds",_damage,_typeOfDamage);
+        continue
+    };
+    private _dmgPerWound = _damage/_nWounds;
+
+    // find the available injuries for this damage type and damage amount
+    private _weightedWoundTypes = [];
+    {
+        private _weighting = _x select 1;
+        private _woundWeight = [_dmgPerWound, _weighting] call FUNC(interpolatePoints);
+        _weightedWoundTypes pushBack _x;
+        _weightedWoundTypes pushBack _woundWeight;
+    } forEach _damageWoundDetails;
+
+    if (_weightedWoundTypes isEqualTo []) then {
+        TRACE_2("No valid wounds",_damage,_typeOfDamage);
+        continue
+    };
 
-            private _bodyPartNToAdd = if (_isSelectionSpecific) then {_bodyPartN} else {selectRandomWeighted _damageSelectionArray};
+    for "_i" from 1 to _nWounds do {
 
-            _bodyPartDamage set [_bodyPartNToAdd, (_bodyPartDamage select _bodyPartNToAdd) + _woundDamage];
-            _bodyPartVisParams set [[1,2,3,3,4,4] select _bodyPartNToAdd, true]; // Mark the body part index needs updating
+        // Select the injury we are going to add
+        selectRandomWeighted _weightedWoundTypes params ["_woundTypeToAdd", "", "_dmgMultiplier", "_bleedMultiplier", "_sizeMultiplier", "_painMultiplier", "_fractureMultiplier"];
+        GVAR(woundDetails) get _woundTypeToAdd params ["","_injuryBleedingRate","_injuryPain","_causeLimping","_causeFracture"];
+        private _woundClassIDToAdd = GVAR(woundClassNames) find _woundTypeToAdd;
+    
+        // Add a bit of random variance to wounds
+        private _woundDamage = _dmgPerWound * _dmgMultiplier * random [0.9, 1, 1.1];
 
+        _bodyPartDamage set [_bodyPartNToAdd, (_bodyPartDamage select _bodyPartNToAdd) + _woundDamage];
+        _bodyPartVisParams set [[1,2,3,3,4,4] select _bodyPartNToAdd, true]; // Mark the body part index needs updating
 
-            // Damage to limbs/head is scaled higher than torso by engine
-            // Anything above this value is guaranteed worst wound possible
-            private _worstDamage = [2, 1, 4, 4, 4, 4] select _bodyPartNToAdd;
+        // Damage to limbs/head is scaled higher than torso by engine
+        // Anything above this value is guaranteed worst wound possible
+        private _worstDamage = [2, 1, 4, 4, 4, 4] select _bodyPartNToAdd;
 
-            // More wounds means more likely to get nasty wound
-            private _countModifier = 1 + random(_i - 1);
+        // Config specifies bleeding and pain for worst possible wound
+        // Worse wound correlates to higher damage, damage is not capped at 1
+        private _woundSize = linearConversion [0.1, _worstDamage, _woundDamage * _sizeMultiplier, 0.25, 1, true];
```

Comment:
I don't feel great about the minimum size value being 0.25. This makes lower-damage wounds proportionally more dangerous (in the worst case, 1/40 the damage results in 1/4 the bleeding) which has a nasty interaction with the thing I mentioned above. Combined with the fact that creating multiple wounds is already punishing because it usually means more work to bandage, this makes controlling the lethality of explosives really complicated and kind of arbitrary.

Reply:
Minimum size value is now determined by the thresholds